*** a/scripts/fetchPufferfish.sh	Tue Sep 17 14:15:07 2024
--- b/scripts/fetchPufferfish.sh	Mon Jun 26 20:51:20 2023
*************** fi
*** 57,63 ****
  rm -fr ${EXTERNAL_DIR}/pufferfish
  unzip ${EXTERNAL_DIR}/pufferfish.zip -d ${EXTERNAL_DIR}
  mv ${EXTERNAL_DIR}/pufferfish-${SVER} ${EXTERNAL_DIR}/pufferfish
! 
  mkdir -p ${INSTALL_DIR}/include/pufferfish
  
  cp ${EXTERNAL_DIR}/pufferfish/include/ProgOpts.hpp ${INSTALL_DIR}/include/pufferfish
--- 57,63 ----
  rm -fr ${EXTERNAL_DIR}/pufferfish
  unzip ${EXTERNAL_DIR}/pufferfish.zip -d ${EXTERNAL_DIR}
  mv ${EXTERNAL_DIR}/pufferfish-${SVER} ${EXTERNAL_DIR}/pufferfish
! sed -i 's/#elif defined(__aarch64__)/#elif 0/g' ${EXTERNAL_DIR}/pufferfish/include/FastxParserThreadUtils.hpp
  mkdir -p ${INSTALL_DIR}/include/pufferfish
  
  cp ${EXTERNAL_DIR}/pufferfish/include/ProgOpts.hpp ${INSTALL_DIR}/include/pufferfish
